api/v1beta1/foundationdbcluster_types.go

```diff
@@ -2107,13 +2108,34 @@ const (
 	RedundancyModeUnset RedundancyMode = ""
 )
 
+// StorageEngine defines the storage engine for the database
+// +kubebuilder:validation:MaxLength=100
+type StorageEngine string
+
+const (
+	// StorageEngineSSD defines the storage engine ssd.
+	StorageEngineSSD StorageEngine = "ssd"
+	// StorageEngineSSD2 defines the storage engine ssd-2.
+	StorageEngineSSD2 StorageEngine = "ssd-2"
+	// StorageEngineMemory defines the storage engine memory.
+	StorageEngineMemory StorageEngine = "memory"
+	// StorageEngineMemory2 defines the storage engine memory-2.
+	StorageEngineMemory2 StorageEngine = "memory-2"
```

Comment:
how about rocksdb engine for master branch?

Reply:
These are only constant used in the code the actual validation fields are here and include the rocksdb engine: https://github.com/FoundationDB/fdb-kubernetes-operator/blob/main/api/v1beta1/foundationdbcluster_types.go#L2136